Gopkg.toml

```diff
@@ -69,3 +69,7 @@
 [[constraint]]
   name = "github.com/golang/protobuf"
   version = "1.2.0"
+
+[[constraint]]
+  name = "github.com/ipfs/go-ipfs-api"
+  branch = "master"
```

Comment:
Is it possible to stick to some specific release version? Sticking to master can lead to future build is broken because of incompatible change in IPFS master branch.

Reply:
shell.OnlyHash(true), this option is available only in the master build. This functionality is new and doesn't available in release versions yet. 

there is no other means to generate the IPFS hash on the client side using plain content, because IPFS hash is not only the hash of the content, it is the combination of multiple hashes, thats why it is called multihash. IPFS gateway determines this hash when it splits the contents and sends to nodes. each piece hash is calculated separately. And IPFS hash is the Hash Addition of all the individual hashes on different nodes. This final used as content ID (cid), which we used as file address in IPFS.

Reply:
ok, then